src/mpi/init/init_async.c

```diff
@@ -59,22 +73,117 @@ static void progress_fn(void *data)
     return;
 }
 
+MPL_STATIC_INLINE_PREFIX int MPIDI_parse_progress_thread_affinity(int *thread_affinity,
+                                                                  int threads_per_node)
+{
+    int th_idx, read_count = 0, mpi_errno = MPI_SUCCESS;
+    char *affinity_copy = NULL;
+    const char *affinity_to_parse = MPIR_CVAR_CH4_PROGRESS_THREAD_AFFINITY;
+    char *proc_id_str, *tmp;
+    size_t proc_count;
+
+    if (!affinity_to_parse || strlen(affinity_to_parse) == 0) {
+        /* generate default affinity */
+        proc_count = MPL_get_nprocs();
+        for (th_idx = 0; th_idx < threads_per_node; th_idx++) {
+            if (th_idx < proc_count)
+                thread_affinity[th_idx] = proc_count - th_idx - 1;
+            else
+                thread_affinity[th_idx] = thread_affinity[th_idx % proc_count];
+        }
+        goto fn_exit;
+    }
+
+    /* create copy of original buffer because it will be modified in strsep */
+    affinity_copy = MPL_strdup(affinity_to_parse);
+    MPIR_Assert(affinity_copy);
+    tmp = affinity_copy;
+
+    for (th_idx = 0; th_idx < threads_per_node; th_idx++) {
+        proc_id_str = MPL_strsep(&tmp, " \n\t,");
```

Comment:
I think the extra flexibility (separate by blank or newline) is more trouble that its worth. How about just use comma?

Reply:
I agree, comma should be sufficient

Reply:
Changed it to take just comma.